xla/pjrt/mlir_to_hlo.cc

```diff
@@ -265,12 +265,12 @@ StatusOr<mlir::OwningOpRef<mlir::ModuleOp>> ParseMlirModuleString(
 
 Status ParseMlirModuleStringAndConvertToXlaComputation(
     absl::string_view mlir_module_str, XlaComputation& xla_computation,
-    bool use_tuple_args, bool return_tuple) {
+    bool use_tuple_args, bool return_tuple, bool legalize_sparse_ops) {
   mlir::MLIRContext context;
   TF_ASSIGN_OR_RETURN(mlir::OwningOpRef<mlir::ModuleOp> module,
                       xla::ParseMlirModuleString(mlir_module_str, context));
   return xla::MlirToXlaComputation(*module, xla_computation, use_tuple_args,
-                                   return_tuple);
+                                   return_tuple, legalize_sparse_ops);
```

Comment:
Looking at the code I think we should go the other direction and remove this flag. It appears to be unused in the `xla::MlirToXlaComputation` implementation?

https://github.com/openxla/xla/blob/e416d4fa6c28ef61486d7b4c1772a51c4abd8bab/xla/pjrt/mlir_to_hlo.cc#L182

Reply:
1. sure, I removed `legalize_sparse_ops` param from `xla::MlirToXlaComputation`.

2. `options` param has default value `{}` in MlirToXlaComputation decl and it is passed by value. In case a func caller uses empty options we can simply omit passing `options` param to remove unnecessary obj copy.